src/main/java/com/google/sps/utils/UserUtils.java

```diff
@@ -47,25 +47,43 @@
     * @param  entityPropertyValue  the value of the property that is filtered for
     * @return                      a single entity that has the same value for the property in the parameter 
     */
+
+public static Entity initializeUser(String userId, String name, String sessionID, DatastoreService datastore){
+        Entity userEntity = getEntityFromDatastore("user", "userID", userId, datastore);
+        if(userEntity == null){
```

Comment:
Branching like this makes me nervous. The behavior of `initializeUser` is completely different, based on whether the user exists yet or not. That reduces the risk of overwriting user data... but it also means that after calling initializeUser, the caller has no idea what the state of the user is.

I'd rather the caller be responsible for checking whether the user has been initialized yet (ie whether their ID is present in datastore). What do you think?

Reply:
I think thats fine, plus it makes more sense with what the function is named. I'll change the implementation